server/openapi_server/controllers/text_date_annotation_controller.py

```diff
@@ -33,6 +36,11 @@ def create_text_date_annotations():  # noqa: E501
                 "(-)(19[0-9][0-9]|20[0-9][0-9])", note._text)
             add_date_annotation(annotations, matches, "MM-DD-YYYY")
 
+            matches = re.finditer(
+                "(19[0-9][0-9]|20[0-9][0-9])-([0-1][0-9])-([0-9][0-9])",
```

Comment:
@gkowalski Why is the regex for "YYYY-MM-DD" smaller than the regex of "MM-DD-YYYY"?

Reply:
Better engineered, more concise.

Reply:
What does that mean? Can you apply your "better engineering" to the other regex?

Reply:
Your regex does not catch `1912-1-1`. This PR has been opened for quite some time now so I'll go ahead and close it.